File: app/src/debug/java/com/example/PerBuildComponentProvider.java
package com.example;

import com.facebook.stetho.okhttp3.StethoInterceptor;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

import static okhttp3.logging.HttpLoggingInterceptor.Level.BODY;

class PerBuildComponentProvider {
  private static PerBuildComponentProvider instance;

  static PerBuildComponentProvider getInstance() {
    if (instance == null) {
      instance = new PerBuildComponentProvider();
    }

    return instance;
  }

  OkHttpClient.Builder okHttpBuilder() {
    return new OkHttpClient.Builder()
      .addNetworkInterceptor(new StethoInterceptor())
      .addInterceptor(new HttpLoggingInterceptor().setLevel(BODY));
  }
}

File: app/src/debug/kotlin/com/example/PerBuildComponentProvider.kt
package com.example

import com.facebook.stetho.okhttp3.StethoInterceptor

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

import okhttp3.logging.HttpLoggingInterceptor.Level.BODY

internal class PerBuildComponentProvider {
  fun okHttpBuilder(): OkHttpClient.Builder {
    return OkHttpClient.Builder()
      .addNetworkInterceptor(StethoInterceptor())
      .addInterceptor(HttpLoggingInterceptor().setLevel(BODY))
  }

  companion object {
    private var instance: PerBuildComponentProvider? = null

    fun getInstance(): PerBuildComponentProvider {
      if (instance == null) {
        instance = PerBuildComponentProvider()
      }

      return instance as PerBuildComponentProvider
    }
  }
}
